Spellcheck not available with latest Arch Linux release.

geany 1.33 (built on 2018-03-06 with GTK 3.22.28, GLib 2.54.3)
geany-plugins 1.33-1
aspell 0.60.6.1-5
enchant 2.2.3-1
hunspell 1.6.2-1
hunspell-en 2017.01.22-2
hunspell-en_US 2017.01.22-2
linux 4.15.8-1-ARCH

After a full system update, the Spellcheck plugin is no longer in the Plugin Manager dialog.

I faced the same issue on Archlinux. I got this error when I forced --enable-spellcheck:
```
checking for enchant >= 1.5... no
checking for enchant >= 2.0... no
checking for enchant >= 1.3... no
configure: error: Package requirements (enchant >= 1.3) were not met:
No package 'enchant' found

```
Note that my current version is `enchant 2.2.3-1`
As a quick workaround, it is enough to install an enchant package which has the version number in the name (eg. enchant2). Such package exists in the AUR, unfortunately, with version 1.6. On the good side, it doesn't conflict with the main enchant package and geany's spellcheck is back (I assume it is using the 1.6 version since it was compiled based on it, please confirm this).  
Looking forward to the definitive fix for this problem.

--- a/build/spellcheck.m4
+++ b/build/spellcheck.m4
@@ -2,17 +2,34 @@ AC_DEFUN([GP_CHECK_SPELLCHECK],
 [
     GP_ARG_DISABLE([spellcheck], [auto])
 
+    ENCHANT_PACKAGE_NAME=enchant-2
     ENCHANT_VERSION=1.3
     ENCHANT_VERSION_1_5=1.5
     ENCHANT_VERSION_2_0=2.0
-    PKG_CHECK_MODULES([ENCHANT_1_5], [enchant >= ${ENCHANT_VERSION_1_5}],
-                      have_enchant_1_5=yes,
-                      have_enchant_1_5=no)
-    PKG_CHECK_MODULES([ENCHANT_2_0], [enchant >= ${ENCHANT_VERSION_2_0}],
-                      have_enchant_2_0=yes,
-                      have_enchant_2_0=no)
+    ENCHANT_VERSION_2_2=2.2
+
+    # check for enchant package
+    PKG_CHECK_MODULES([ENCHANT_2_2], [${ENCHANT_PACKAGE_NAME} >= ${ENCHANT_VERSION_2_2}],
+                      have_enchant_2_2=yes,
+                      have_enchant_2_2=no)
+    if [[ x"$have_enchant_2_2" = "xyes" ]]; then
+        # we have got the new enchant-2 package
+        have_enchant_1_5=yes
+        have_enchant_2_0=yes
+    else
+        # check for old enchant package
+        PKG_CHECK_MODULES([ENCHANT_1_5], [enchant >= ${ENCHANT_VERSION_1_5}],
+                          have_enchant_1_5=yes,
+                          have_enchant_1_5=no)
+        PKG_CHECK_MODULES([ENCHANT_2_0], [enchant >= ${ENCHANT_VERSION_2_0}],
+                          have_enchant_2_0=yes,
+                          have_enchant_2_0=no)
+
+        ENCHANT_PACKAGE_NAME=enchant
+    fi
+
     GP_CHECK_PLUGIN_DEPS([spellcheck], [ENCHANT],
-                         [enchant >= ${ENCHANT_VERSION}])
+                         [${ENCHANT_PACKAGE_NAME} >= ${ENCHANT_VERSION}])
 
     AM_CONDITIONAL([HAVE_ENCHANT_1_5], [test "$have_enchant_1_5" = yes])
     AM_CONDITIONAL([HAVE_ENCHANT_2_0], [test "$have_enchant_2_0" = yes])

